src/axom/primal/tests/primal_clip.cpp

```diff
@@ -1294,6 +1294,179 @@ TEST(primal_clip, tet_tet_clip_special_case_1)
+{
+  using namespace Primal3D;
+  constexpr double EPS = 1e-10;
+  constexpr bool CHECK_SIGN = true;
+
+  // Plane intersects one edge of tet
+  TetrahedronType tet(PointType {0.0, 0.0, 0.0},
+                      PointType {1.0, 0.0, 0.0},
+                      PointType {0.0, 1.0, 0.0},
+                      PointType {0.0, 0.0, 1.0});
+
+  PlaneType plane(VectorType {0.0, -1.0, -1.0}, 0.0);
+
+  PolyhedronType poly = axom::primal::clip(tet, plane, EPS, CHECK_SIGN);
+
+  EXPECT_NEAR(0.0, poly.volume(), EPS);
+}
+
+TEST(primal_clip, tet_plane_border_edge_above)
+{
+  using namespace Primal3D;
+  constexpr double EPS = 1e-10;
+  constexpr bool CHECK_SIGN = true;
+
+  // Plane intersects one edge of tet
+  TetrahedronType tet(PointType {0.0, 0.0, 0.0},
+                      PointType {1.0, 0.0, 0.0},
+                      PointType {0.0, 1.0, 0.0},
+                      PointType {0.0, 0.0, 1.0});
+
+  PlaneType plane(VectorType {0.0, 1.0, 1.0}, 0.0);
+
+  PolyhedronType poly = axom::primal::clip(tet, plane, EPS, CHECK_SIGN);
+
+  EXPECT_NEAR(tet.volume(), poly.volume(), EPS);
+}
+
+TEST(primal_clip, tet_plane_border_face_below)
+{
+  using namespace Primal3D;
+  constexpr double EPS = 1e-10;
+  constexpr bool CHECK_SIGN = true;
+
+  // Tet and plane do not intersect, but border each other
+  TetrahedronType tet(PointType {0.0, 0.0, 0.0},
+                      PointType {1.0, 0.0, 0.0},
+                      PointType {0.0, 1.0, 0.0},
+                      PointType {0.0, 0.0, 1.0});
+
+  PlaneType plane(VectorType {0.0, 0.0, -1.0}, 0.0);
+
+  PolyhedronType poly = axom::primal::clip(plane, tet, EPS, CHECK_SIGN);
+
+  EXPECT_NEAR(0.0, poly.volume(), EPS);
+}
+
+TEST(primal_clip, tet_plane_border_face_above)
+{
+  using namespace Primal3D;
+  constexpr double EPS = 1e-10;
+  constexpr bool CHECK_SIGN = true;
+
+  // Tet and plane do not intersect, but border each other
+  TetrahedronType tet(PointType {0.0, 0.0, 0.0},
+                      PointType {1.0, 0.0, 0.0},
+                      PointType {0.0, 1.0, 0.0},
+                      PointType {0.0, 0.0, 1.0});
+
+  PlaneType plane(VectorType {0.0, 0.0, 1.0}, 0.0);
+
+  PolyhedronType poly = axom::primal::clip(plane, tet, EPS, CHECK_SIGN);
+
+  EXPECT_NEAR(tet.volume(), poly.volume(), EPS);
+}
+
+TEST(primal_clip, tet_plane_intersect_three_edges)
+{
+  using namespace Primal3D;
+  constexpr double EPS = 1e-10;
+  constexpr bool CHECK_SIGN = true;
+
+  // Plane intersects three edges of tet
+  TetrahedronType tet(PointType {0.0, 0.0, 0.0},
+                      PointType {1.0, 0.0, 0.0},
+                      PointType {0.0, 1.0, 0.0},
+                      PointType {0.0, 0.0, 1.0});
+
+  PlaneType plane(VectorType {-1.0, 0.0, 1.0}, 0.0);
+
+  PolyhedronType poly = axom::primal::clip(plane, tet, EPS, CHECK_SIGN);
+
+  EXPECT_NEAR(tet.volume()/2.0, poly.volume(), EPS);
+}
+
+// TODO: Add a test for a plane that intersects four edges of a tet
```

Comment:
👍 -- that would be a nice additional test.

One possibility would be to define a tet w/ vertices at opposite corners of a unit cube, and a plane that bisects the cube.

I think something like:
```cpp
  TetrahedronType tet(PointType {0.0, 0.0, 0.0},
                      PointType {1.0, 1.0, 0.0},
                      PointType {0.0, 1.0, 1.0},
                      PointType {1.0, 0.0, 1.0});
  PlaneType plane(VectorType {0.0, 0.0, 1.0}, 0.5);
```
although you might need to play with the orientations.

By symmetry, I'd expect the clipped volume to be `tet.volume()/2`

Reply:
That one worked! Thanks!